ykrt/src/compile/jitc_yk/jit_ir.rs

```diff
@@ -20,6 +20,27 @@ use ykaddr::addr::symbol_vaddr;
 // This is simple and can be shared across both IRs.
 pub(crate) use super::aot_ir::Predicate;
 
+/// The symbol name of the global variable pointers array.
+///
+/// FIXME: At the moment we have to use `dlsym` to look this variable up, because it will only be
+/// present if `YKD_NEW_CODEGEN=1`. When the new codegen becomes the only codegen, we can move from
+/// `dlsym` to:
+///
+/// ```rust
+/// extern "C" {
+///     /// The address of this variable (note: not the contents of the variable!) points to the
+///     /// beginning of an array of `usize`s.
+///     static __yk_globalvar_ptrs: usize;
+///     /// How many elements are there in `__yk_globalvar_ptrs`?
+///     static __yk_globalvar_len: i64;
```

Comment:
I'm OK with the `extern "C"` stuff as long as we are OK with hacking/mocking around it being absent in Rust unit tests.

I still think that `__yk_globalvar_len` can be removed, and then the compiler change isn't needed any more.

Reply:
I tend to agree and I will probably close this PR and its ykllvm companion. Juggling other branches first, though.